oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/MapReducer.java

```diff
@@ -754,17 +755,57 @@ protected MapReducer(MapReducer<?> obj) {
     // apply geometry filter as first map function
     final List<MapFunction> remainingMappers = List.copyOf(ret.mappers);
     ret.mappers.clear();
-    if (ret.forClass.equals(OSMContribution.class)) {
-      ret = ret.filter(x -> {
-        OSMContribution c = (OSMContribution) x;
-        return f.applyOSMGeometry(c.getEntityBefore(), c::getGeometryBefore)
-          || f.applyOSMGeometry(c.getEntityAfter(), c::getGeometryAfter);
-      });
-    } else if (ret.forClass.equals(OSMEntitySnapshot.class)) {
-      ret = ret.filter(x -> {
-        OSMEntitySnapshot s = (OSMEntitySnapshot) x;
-        return f.applyOSMGeometry(s.getEntity(), s::getGeometry);
-      });
+    if (this.grouping == Grouping.NONE) {
+      // no grouping -> directly filter using the geometries of the snapshot / contribution
+      if (ret.forClass.equals(OSMEntitySnapshot.class)) {
+        ret = ret.filter(x -> {
+          OSMEntitySnapshot s = (OSMEntitySnapshot) x;
+          return f.applyOSMGeometry(s.getEntity(), s::getGeometry);
+        });
+      } else if (ret.forClass.equals(OSMContribution.class)) {
+        ret = ret.filter(x -> {
+          OSMContribution c = (OSMContribution) x;
+          if (c.is(ContributionType.CREATION)) {
+            return f.applyOSMGeometry(c.getEntityAfter(), c::getGeometryAfter);
+          } else if (c.is(ContributionType.DELETION)) {
+            return f.applyOSMGeometry(c.getEntityBefore(), c::getGeometryBefore);
+          } else {
+            return f.applyOSMGeometry(c.getEntityBefore(), c::getGeometryBefore)
+                || f.applyOSMGeometry(c.getEntityAfter(), c::getGeometryAfter);
```

Comment:
According to my local tests, the lines 770 to 774 are still not tested. They look sane to me, so I wouldn't require it, but maybe it makes sense to do that anyway.

Reply:
if we had a properly crafted test data set, it would be easy to trigger these branches as well. That would be nice to have eventually. But I think we can live with the current implementation for now, right?